components/automate-ui/src/app/entities/roles/role.reducer.ts

```diff
@@ -24,7 +24,7 @@ export function roleEntityReducer(state: RoleEntityState = RoleEntityInitialStat
 
   switch (action.type) {
     case RoleActionTypes.GET_ALL:
-      return set('getAllStatus', EntityStatus.loading, roleEntityAdapter.removeAll(state));
+      return set('getAllStatus', EntityStatus.loading, state);
```

Comment:
I tested this out to make sure none of the [data leakage issues](https://github.com/chef/automate/issues/2138) came back with the revert. 
I only saw leakage with roles. To repro, make this policy and log in as the test user:

```
$ echo '{
    "name": "testing",
    "id": "test1",
    "members": [
      "user:local:test"
    ],
    "statements": [
      {
        "effect": "ALLOW",
        "actions": ["iam:*:list"],
        "projects": [
          "*"
        ]
      }
    ],
    "projects": []
  }' >> pol.json

$ curl -kH "api-token: $TOK" -X POST -d @pol.json https://localhost/apis/iam/v2beta/policies | jq
```

If you navigate to roles, then click on one to go to its details page, you'll still be able to see all the role details despite the GET request returning 403

Reply:
That does not quite make sense. The only thing I did was not erase data in the middle of a ListRoles. At the completion of the ListRoles call everything is exactly as it was before. The GetRole call is unchanged. Can discuss further tomorrow. 🤔

Reply:
Yeah looks like this is showing up on master too so must be a separate bug! I'll file it